opm/output/eclipse/EclipseWriter.cpp

```diff
@@ -853,15 +793,29 @@ EclipseWriter::EclipseWriter( std::shared_ptr< const EclipseState > es,
        evaluation of region properties.
     */
     {
-        const auto& properties = es->get3DProperties();
+        const auto& properties = this->impl->es->get3DProperties();
         const auto& fipnum = properties.getIntGridProperty("FIPNUM");
         const auto& region_values = properties.getRegions( "FIPNUM" );
 
         for (auto region_id : region_values)
-            this->impl->regionCells.emplace( region_id , fipnum.cellsEqual( region_id , this->impl->compressedToCartesian ));
+            this->impl->regionCells.emplace( region_id , fipnum.cellsEqual( region_id , this->impl->grid ));
     }
 }
 
+
+EclipseWriter::EclipseWriter(std::shared_ptr< const EclipseState > es, const EclipseGrid& grid)
+    : impl( new Impl( es , grid) )
+{
+    init();
```

Comment:
This needs to go away.

A better idea is actually to only pass the `EclipseGrid` argument by value (since it's being copied anyway) and not overload the constructor. It removes the needs for a common `init`, makes it obvious what's going on, and caller can choose to move if he wants to.